src/libecalc/application/graph_result.py

```diff
@@ -130,22 +150,27 @@ def _evaluate_installations(self, variables_map: dto.VariablesMap) -> List[libec
             ]
 
             installation_node_info = self.graph.get_node_info(installation.id)
-            power = reduce(
-                operator.add,
-                [
-                    TimeSeriesRate.from_timeseries_stream_day_rate(component.power, regularity=regularity)
-                    for component in sub_components
-                    if self.graph.get_node_info(component.id).component_type == ComponentType.GENERATOR_SET
-                ],
-                TimeSeriesRate(
-                    values=[0.0] * self.variables_map.length,
-                    timesteps=self.variables_map.time_vector,
-                    unit=Unit.MEGA_WATT,
-                    rate_type=RateType.STREAM_DAY,
-                    regularity=regularity.values,
-                ),  # Initial value, handle no power output from components
+
+            power_electrical = self._compute_aggregated_power(
+                sub_components=sub_components, regularity=regularity, components=[ComponentType.GENERATOR_SET]
             )
 
+            fuel_consumer_nodes = self.graph.get_fuel_consumer_nodes()
+            fuel_consumer_sub_components = [
+                sub_component for sub_component in sub_components if sub_component.id in fuel_consumer_nodes
+            ]
+
+            power_mechanical = self._compute_aggregated_power(
+                sub_components=fuel_consumer_sub_components,
+                regularity=regularity,
+                components=[ComponentType.COMPRESSOR, ComponentType.COMPRESSOR_V2],
+            )
```

Comment:
Will this work if we have a compressor that consumes electrical power?

Reply:
Yes, because we have ensured that only fuel consumers are sent in (fuel_consumer_sub_components)

Reply:
Have changed to consumption type instead